README.md

```diff
@@ -201,52 +201,37 @@ click options or function keywords:
 
 ### Advanced: adding arguments to built-in commands
 
-Instead of rewriting a command from scratch, a project may want to add a flag to a built-in `spin` command, or perhaps do some pre- or post-processing.
-For this, we have to use an internal Click concept called a [context](https://click.palletsprojects.com/en/8.1.x/complex/#contexts).
-Fortunately, we don't need to know anything about contexts other than that they allow us to execute commands within commands.
+Instead of rewriting a command from scratch, a project may simply want to add a flag to an existing `spin` command, or perhaps do some pre- or post-processing.
+For this purpose, we provide the `spin.util.extend_cmd` decorator.
 
-We proceed by duplicating the function header of the existing command, and adding our own flag:
+Here, we show how to add a `--extra` flag to the existing `build` function:
 
 ```python
 from spin.cmds import meson
 
-# Take this from the built-in implementation, in `spin.cmds.meson.build`:
 
+@click.option("-e", "--extra", help="Extra test flag")
+@util.extend_command(spin.cmds.meson.build)
```

Comment:
maybe make this `spin.util.extend_command` or an explicit import to this example?